miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt

```diff
@@ -36,10 +36,15 @@ internal class ApiClient @VisibleForTesting constructor(
     )
 
     suspend fun list(): List<MiniAppInfo> {
-        val request = listingApi.list(hostAppId, hostAppVersionId)
+        val request = appInfoApi.list(hostAppId, hostAppVersionId)
         return requestExecutor.executeRequest(request)
     }
 
+    suspend fun fetchInfo(appId: String): MiniAppInfo {
+        val request = appInfoApi.fetchInfo(hostAppId, hostAppVersionId, appId)
+        return requestExecutor.executeRequest(request).first()
```

Comment:
Instead of returning a single object, the backend returns a list -> it looks painful to me 😂
1/ `List<T>.first()` throws an exception for empty list. May you want to handle it with our sdk exception?

2/ We assume that there is only one item in the response list but let's create a unit test for the case: `should only get the first item from mini app list response`

Reply:
The first part I agree with, but from backend, for now, we have to face the list.
The other two have been updated :) 👍